pandas/core/arrays/boolean.py

```diff
@@ -317,18 +313,18 @@ def map_string(s):
         scalars = [map_string(x) for x in strings]
         return cls._from_sequence(scalars, dtype, copy)
 
-    def _values_for_factorize(self) -> Tuple[np.ndarray, Any]:
+    def _values_for_factorize(self) -> Tuple[np.ndarray, int]:
```

Comment:
In general, though, the "Any" is correct (I don't know what the typing should do here, but the signature of the base class would use "Any", so any place where `_values_for_factorize` is called would need to assume "Any")